messaging-client-databus/src/main/java/com/inmobi/databus/readers/LocalStreamCollectorReader.java

```diff
@@ -281,4 +281,63 @@ protected Date getTimeStampFromCollectorStreamFile(FileStatus file) {
     }
     return null;
   }
+
+  protected Long doRecursiveSizing(Path dir, PathFilter pathFilter) throws Exception {
+    Long pendingSize =0l;
+    FileStatus[] fileStatuses = fsListFileStatus(dir, pathFilter);
+    if (fileStatuses == null || fileStatuses.length == 0) {
+      LOG.debug("No files in directory:" + dir);
+    } else {
+      for (FileStatus file : fileStatuses) {
+        if (file.isDir()) {
+          doRecursiveSizing(file.getPath(), pathFilter);
+        } else {
+          try {
+            Date currentTimeStamp = LocalStreamCollectorReader.
+                    getDateFromStreamFile(streamName, file.getPath().getName());
+            if (stopTime != null && stopTime.before(currentTimeStamp)) {
+              stopListing();
```

Comment:
why call stopListing()? This will have unintended side-effects. Instead just continue in the for loop.